app/util/SwitchLayer.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * Util class for switchLayer related functions.
+ *
+ * @class CpsiMapview.util.SwitchLayer
+ */
+Ext.define('CpsiMapview.util.SwitchLayer', {
+    alternateClassName: 'SwitchLayerUtil',
+    singleton: true,
+
+    /** The possible states of the layer switch */
+    switchStates: {
+        ABOVE_SWITCH_RESOLUTION : 'cmv_above_switch_resolution',
+        BELOW_SWITCH_RESOLUTION : 'cmv_below_switch_resolution'
+    },
+
+    /**
+     * Loops through all layers, identifies switch layers
+     * and replaces them if required
+     *
+     * @param {Number} resolution The new resolution
+     */
+    handleSwitchLayerOnResolutionChange: function(resolution) {
+        var layerCollection = BasiGX.util.Map.getMapComponent().getMap().getLayers();
+
+        this.checkSwitchLayersRecursively(layerCollection, resolution);
```

Comment:
Please use `var staticMe = CpsiMapview.util.SwitchLayer;` instead of `this` like in the other functions.

Reply:
@chrismayer thanks for the hint, I fixed it